src/UnitTests/IMappingExpression/ForCtorParam.cs

```diff
@@ -51,5 +55,52 @@ public void Should_resolve_using_custom_func()
 
             dest.Value1.ShouldEqual(8);
         }
+
+        [Fact]
+        public void Should_ignore_nonexistent_parameter()
+        {
+            var config = new MapperConfiguration(cfg =>
+            {
+                cfg.CreateMap<Source, Dest>()
+                    .ForCtorParam("thing", opt => opt.MapFrom(src => src.Value))
+                    .ForCtorParam("think", opt => opt.MapFrom(src => src.Value));
```

Comment:
Do you mean something like `public Dest(string foo, int bar)`? The same thing happens - the mismatching configuration is ignored, and the rest work just as if it wasn't there. The mapping works if all parameters are mapped, and doesn't if they're not. Or was that not what you were asking?